safe-fndn/safe-singleton-factory
# Introduce Utility to Deploy From GitHub Issue

This PR adds a new `github-deploy.sh` script that can be used to deploy the Safe singleton factory from a GitHub issue.

Sample GH PR created with script: https://github.com/safe-global/safe-singleton-factory/pull/454

### Test Plan

While the private key is confidential, you can still test this script against a local network with an alternative account. For this some hidden magical `GITHUB_DEPLOY_...` environment variables were added for local testing:

1. Create a new 1Password item and get its ID; you must ensure that it has a `password` field that is the mnemonic of the account to use for deployment (which must have sufficient funds).
2. Run a local development node (tip: you can use `anvil` and the `test test ... test junk` mnemonic).
3. run `GITHUB_DEPLOY_ITEMUID=... bash bin/github-deploy.sh 472`

This will run the entire script **except** actually publishing the branch and creating the PR with the `gh` tool for testing purposes.
